packages/edit-post/src/components/options-modal/meta-boxes-section.js

```diff
@@ -34,12 +34,13 @@ export function MetaBoxesSection( { areCustomFieldsRegistered, metaBoxes, ...sec
 }
 
 export default withSelect( ( select ) => {
-	const { getEditorSettings } = select( 'core/editor' );
+	const { getPostType } = select( 'core' );
+	const { getEditedPostAttribute } = select( 'core/editor' );
 	const { getAllMetaBoxes } = select( 'core/edit-post' );
+	const postType = getPostType( getEditedPostAttribute( 'type' ) );
 
 	return {
-		// This setting should not live in the block editor's store.
-		areCustomFieldsRegistered: getEditorSettings().enableCustomFields !== undefined,
```

Comment:
What's the purpose of the previous `enableCustomFields` setting?
I assume it's meant to be set in WordPress Core. is there code we need to remove from there?

Reply:
`supportsCustomFields` defines if the post type supports this feature. `enableCustomFields` is the setting if this meta-box is visible. Before `enableCustomFields` had the default value `false` we checked with `!== undefined` if the feature exits. Now `enableCustomFields` can't be `undefined` anymore, so we need to check `supportsCustomFields`.

We still need `enableCustomFields` for checking if the meta-box should be visible. This logic is already merged into core.

Reply:
Ok, makes sense. Changes look good to me. I'll wait for @talldan or @noisysocks to confirm.